src/app/lib/business/dad_jokes/dad_jokes_service.dart

```diff
@@ -11,8 +11,8 @@ import 'package:rxdart/rxdart.dart';
 /// Interface for the dad jokes service.
 abstract interface class DadJokesService implements Disposable {
   factory DadJokesService(
-    DadJokesRepository dadJokesRepository,
-    FavoriteDadJokesRepository favoriteDadJokesRepository,
+    DadJokesRepositoryDecorator dadJokesRepository,
```

Comment:
This should probably stay like it was, but the decorator should implement the interface and have the inner service. Like on one of our internal projects right?

Reply:
I was under the impression that flutter wouldn't recognize the decorator as the repository because they don't extend they implement the interface but it seems the service understands that they have the same parent so you was right and I learned something lol

Reply:
Yeah the business service must not know about the specific implementations. It needs to use only the abstractions. Otherwise it gets tightly coupled to a specific implementation and breaks the possibility of substitution.